File: backend/src/main/java/com/whatcanicook/dto/request/CreateRecipeRequest.java
package com.whatcanicook.dto.request;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateRecipeRequest {

    private String title;

    private String description;

    private boolean publicRecipe = true;

    private Long userId;

    private List<Long> ingredientIds;

    private List<String> steps;
}

File: frontend/app/src/main/java/com/app/dto/request/CreateRecipeRequest.kt
package com.app.dto.request

data class CreateRecipeRequest(

    val title: String,

    val description: String,

    val publicRecipe: Boolean,

    val userId: Long,

    val ingredientIds: List<Long> = emptyList(),

    val steps: List<String> = emptyList()
)
